src/Storage/Builder.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Storage;
+
+use BadMethodCallException;
+use Lcobucci\Jose\Parsing;
+use Lcobucci\JWT\Builder as BuilderInterface;
+use Lcobucci\JWT\Signer;
+use Lcobucci\JWT\Signer\Key;
+use Lcobucci\JWT\Token as TokenInterface;
+
+/**
+ * This class makes easier the token creation process
+ *
+ * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
+ * @since 0.1.0
+ */
+final class Builder implements BuilderInterface
+{
+    /**
+     * The token header
+     *
+     * @var array
+     */
+    private $headers;
+
+    /**
+     * The token claim set
+     *
+     * @var array
+     */
+    private $claims;
+
+    /**
+     * The token signature
+     *
+     * @var string|null
+     */
+    private $signature;
+
+    /**
+     * The data encoder
+     *
+     * @var Parsing\Encoder
+     */
+    private $encoder;
+
+    /**
+     * Initializes a new builder
+     */
+    public function __construct(Parsing\Encoder $encoder)
+    {
+        $this->encoder = $encoder;
+        $this->headers = ['typ'=> 'JWT', 'alg' => 'none'];
+        $this->claims = [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canOnlyBeUsedBy(string $audience, bool $addHeader = false): BuilderInterface
+    {
+        $audiences = $this->claims['aud'] ?? [];
+        $audiences[] = $audience;
+
+        return $this->setRegisteredClaim(
+            'aud',
+            array_values(array_map('strval', $audiences)),
```

Comment:
Actually I'll solve it here, that `array_map() + array_values()` is useless now...